src/disco/containers/InstallButton.js

```diff
@@ -0,0 +1,63 @@
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+import { gettext as _ } from 'core/utils';
+
+import 'disco/css/InstallButton.scss';
+import {
+  DOWNLOADING,
+  INSTALLED,
+  UNKNOWN,
+  validInstallStates as validStates,
+} from 'disco/constants';
+
+export class InstallButton extends React.Component {
+  static propTypes = {
+    handleClick: PropTypes.func,
+    handleChange: PropTypes.func,
+    guid: PropTypes.string,
+    progress: PropTypes.number,
+    slug: PropTypes.string.isRequired,
+    state: PropTypes.oneOf(validStates),
+  }
+
+  static defaultProps = {
+    state: UNKNOWN,
```

Comment:
On further reflection maybe this should be component state and not a prop since it is changed based on user events. 

But in which case should we actually have a dumb component for the UI wrapped by a stateful component instead. That way all the interaction handling should happen in the stateful component and the addon just does deals with what it is handed?

I found this which has a useful re-cap of state vs props: https://github.com/uberVU/react-guide/blob/master/props-vs-state.md#props-vs-state